### PR TITLE
ci: fix build of release assets 

### LINKED ISSUE #764: [BUG] flagd version incorrect for 0.6.0
### Observed behavior

1. Download the `0.6.0` version of `flagd`
2. Extract and run `flagd_linux_x86_64 version` which shows:

```
$ ./flagd_linux_x86_64  version

                 ______   __       ________   _______    ______
                /_____/\ /_/\     /_______/\ /______/\  /_____/\
                \::::_\/_\:\ \    \::: _  \ \\::::__\/__\:::_ \ \
                 \:\/___/\\:\ \    \::(_)  \ \\:\ /____/\\:\ \ \ \
                  \:::._\/ \:\ \____\:: __  \ \\:\\_  _\/ \:\ \ \ \
                   \:\ \    \:\/___/\\:.\ \  \ \\:\_\ \ \  \:\/.:| |
                    \_\/     \_____\/ \__\/\__\/ \_____\/   \____/_/

flagd: dev (HEAD), built at: unknown
```

### Expected Behavior

`flagd version` should show `0.6.0`

### Steps to reproduce

_No response_

### PR DESCRIPTION
**Pull request by thisthat**
### Related Issues

Fixes #764 

### How to test

A darwin arm64 image is available here: https://github.com/thisthat/flagd/actions/runs/5665400408 built at this [commit](https://github.com/thisthat/flagd/commit/484da939c3bd331eda2c4025ab3061bcd591e42c)

image

### CODE REVIEW

**Review by RealAnna — APPROVED** [review]

**Review by mowies — APPROVED** [review]
LGTM but those builds could really use a matrix :)

**Review by odubajDT — APPROVED** [review]

File: .github/workflows/release-please.yaml
[the remark this answers is not in the document]
Comment (thisthat) [reply .github/workflows/release-please.yaml:1]: We indeed use the date generated at that step: https://github.com/open-feature/flagd/pull/767/files#diff-228ba02c3afc2e53cd9a66228adf16ad1132106c567d84ca43f6ce4f717debb0R169

[the remark this answers is not in the document]
Comment (beeme1mr) [reply .github/workflows/release-please.yaml:1]: I'm just saying that you can remove lines 104-106 and update line 124 to use `${{ needs.release-please.outputs.date }}`.

[the remark this answers is not in the document]
Comment (thisthat) [reply .github/workflows/release-please.yaml:1]: oohh! Gotcha! I missed that 👍

**Pull request merged**